webapp/src/components/systemConsole/questions/questions.tsx

```diff
@@ -0,0 +1,147 @@
+            },
+            {
+                id: utils.uuid(),
+                text: 'How likely are you to recommend Mattermost?',
+                type: 'linear_scale',
+                system: true,
+                mandatory: true,
+            },
+            {
+                id: utils.uuid(),
+                text: 'How can we make your experience better?',
+                type: 'text',
+                system: true,
+                mandatory: true,
+            },
+            {
+                id: utils.uuid(),
+                type: 'text',
+                system: false,
+                mandatory: false,
+            },
+        ];
+    };
+
+    useEffect(() => {
+        // Set initial value from saved config
+        const savedQuestions = config.PluginSettings.Plugins?.['com.mattermost.user-survey']?.systemconsolesetting.SurveyQuestions;
+
+        const initialSetting = savedQuestions || generateDefaultQuestions();
+        setQuestions(initialSetting);
+        setInitialSetting(id, initialSetting);
+    }, [config.PluginSettings.Plugins, id, setInitialSetting]);
+
+    const saveSettings = useCallback((setting: SurveyQuestionsConfig) => {
+        setSaveNeeded();
+        onChange(id, setting);
+    }, [id, onChange, setSaveNeeded]);
+
+    const questionOnChangeHandler = useDebouncedCallback(
+        useCallback((e: React.ChangeEvent<HTMLInputElement>, questionID: string) => {
+            const newQuestions = [...questions];
+            const i = questions.findIndex((q) => q.id === questionID);
+            newQuestions[i] = {
+                ...newQuestions[i],
+                text: e.target.value,
+            };
+            setQuestions(newQuestions);
+            saveSettings(newQuestions);
+        }, [questions, saveSettings]),
+        500,
+    );
+
+    const renderQuestions = useMemo(() => {
+        return questions.map((question) => {
+            return (
+                <div
+                    key={question.id}
+                    className='vertical question'
+                >
+                    <span className='questionTitle'>
+                        {question.title && question.title}
+                        {
+                            !question.title &&
+                            (`${questionTypeDisplayName.get(question.type)} ${question.mandatory ? '' : '(Optional)'}`)
+                        }
+                    </span>
+
+                    <input
+                        maxLength={1000}
+                        className={`form-control questionInput ${question.system && 'disabled'}`}
+                        defaultValue={question.text}
+                        onChange={(e) => questionOnChangeHandler(e, question.id)}
+                        disabled={question.system}
+                        placeholder='Question'
+                    />
+
+                    {
+                        question.helpText &&
```

Comment:
Nit: can we put this on the above line? Just to save some space. I know this is for readibility, but this is a bit overkill IMO. 0/5 though.

Reply:
This code is gone now.